vortex-array/src/array/chunked/mod.rs

```diff
@@ -132,6 +133,78 @@ impl ChunkedArray {
     pub fn array_stream(&self) -> impl ArrayStream + '_ {
         ArrayStreamAdapter::new(self.dtype().clone(), stream::iter(self.chunks().map(Ok)))
     }
+
+    pub fn rechunk_default(&self) -> VortexResult<Self> {
+        let sixteen_megabytes = 1 << 24;
+        let chunk_max = 1 << 16; // BtrBlocks uses 64K so we use 64Ki
+        self.rechunk(sixteen_megabytes, chunk_max)
+    }
+
+    pub fn rechunk(&self, max_n_bytes: usize, max_n_elements: usize) -> VortexResult<Self> {
+        fn combine_validities(
+            dtype: &DType,
+            validities: Vec<LogicalValidity>,
+        ) -> VortexResult<Validity> {
+            let validity = validities.clone().into_iter().collect::<Validity>();
+            match (dtype.is_nullable(), validity) {
+                (true, validity) => Ok(validity),
+                (false, Validity::AllValid) => Ok(Validity::NonNullable),
+                (false, _) => vortex_bail!(
+                    "for non-nullable dtype, child validities ought to all be AllValid"
+                ),
+            }
+        }
+
+        let dtype = self.dtype();
+
+        let mut new_chunks: Vec<Array> = Vec::new();
+        let mut validities_to_combine: Vec<LogicalValidity> = Vec::new();
+        let mut chunks_to_combine: Vec<Array> = Vec::new();
+        let mut new_chunk_n_bytes = 0;
+        let mut new_chunk_n_elements = 0;
+        for chunk in self.chunks() {
```

Comment:
I think this could be implemented by slicing yourself and then canonicalizing the sliced amount? You just keep track of number of chunks to slice off when you have enough values you slice, canonicalize and add that to the result chunk array

Reply:
hmm maybe not, let me think more about it